languages/go/exercises/concept/structs/structs.go

```diff
@@ -1,20 +1,33 @@
 package structs
 
-// NewCar creates a new remote controlled car with full battery and given specifications.
-func NewCar(speed, batteryDrain int) Car {
-	panic("Please implement the NewCar function")
-}
-
 // Car implements a remote controlled car.
 type Car struct {
 	speed        int
 	batteryDrain int
-
 	battery  int
 	distance int
 }
 
-// TODO: add race track and instantiator function
+// NewCar creates a new remote controlled car with full battery and given specifications.
+func NewCar(speed, batteryDrain int) Car {
+	return Car{
+		speed:        speed,
+		batteryDrain: batteryDrain,
+		battery:      100,
+	}
+}
+
+
+// Track implements a race track.
+type Track struct{
+	distance int
+}
+
+
+// NewTrack created a new track
+func NewTrack(distance int) Track{
+	panic("Please implement the NewTrack function") 
+}
```

Comment:
My original plan was to let the student create this on their own. They have an example (`Car` and `NewCar`).

Do you think that leaves too much wiggle room for the student?

Reply:
No, it should be. Or we can have only the function declared and let them create the Track type.